cmd/traffic_manager/traffic_manager.cc

```diff
@@ -611,31 +614,35 @@ main(int argc, const char **argv)
 
   /* Update cmd line overrides/environmental overrides/etc */
   if (tsArgs) { /* Passed command line args for proxy */
+    char* new_proxy_opts = new char[strlen(lmgmt->proxy_options) + strlen(tsArgs) + 1];
+    strcpy(new_proxy_opts, lmgmt->proxy_options);
+    strcat(new_proxy_opts, tsArgs);
     ats_free(lmgmt->proxy_options);
-    lmgmt->proxy_options = tsArgs;
-    mgmt_log("[main] Traffic Server Args: '%s'\n", lmgmt->proxy_options);
+    lmgmt->proxy_options = new_proxy_opts;
```

Comment:
At this point ``lmgmt->proxy_options`` isn't set, so just copy ``tsArgs`` into the text buffer if it is present.